be/src/service/service_be/lake_service.cpp

```diff
@@ -450,6 +450,28 @@ void LakeServiceImpl::delete_txn_log(::google::protobuf::RpcController* controll
     latch.wait();
 }
 
+namespace drop_table_helper {
+static bthread::Mutex g_mutex;
+static std::unordered_set<std::string> g_paths;
+
+// Return true if the container doesn't already contain "path".
+bool check_path(const std::string& path) {
+    if (!path.empty()) {
+        std::lock_guard l(g_mutex);
+        return g_paths.insert(path).second;
```

Comment:
is it possible to have a max size of the set, if the set is too large, just skip the check?

Reply:
Because the thread pool for executing tasks has a limited queue size (2048), and when the thread pool queue is full, the task will fail immediately and the path will be deleted from the set immediately, it can be assumed that the maximum size of the set will not exceed the thread pool queue capacity by too much.